### ISSUE TITLE
 Warning: array_replace_recursive(): Argument #1 is not an array in /home/pl   aysports/public_html/magento2/vendor/magento/framework/App/DeploymentConfig   /Writer.php on line 135

### ISSUE THREAD

**Issue by mikeguy** [magento/magento2#14187]
1. PHP 7.1
2. Magneto 2.2.3
4. MYSQL 5.6.38

### Steps to reproduce

1. ` magento module:disable Dotdigitalgroup_Email`
2. `magento setup:di:compile`
3. 

### Expected result
1. Able to compile and get the store running again 

### Actual result
```
[root@server magento2]# bin/magento setup:di:compile
Compilation was started.
Proxies code generation... 0/7 [>---------------------------]   0% < 1 sec 44.0                                                   Proxies code generation... 1/7 [====>-----------------------]  14% 1 sec 48.0 Mi                                                  Repositories code generation... 1/7 [====>-----------------------]  14% 1 sec 48                                                  Repositories code generation... 2/7 [========>-------------------]  28% 3 secs 1                                                  Service data attributes generation... 2/7 [========>-------------------]  28% 3                                                   Service data attributes generation... 3/7 [============>---------------]  42% 3                                                   Application code generator... 3/7 [============>---------------]  42% 3 secs 164                                                  Application code generator... 4/7 [================>-----------]  57% 7 secs 188                                                  Interceptors generation... 4/7 [================>-----------]  57% 7 secs 188.0                                                   MiB

  [Exception]
  Warning: array_replace_recursive(): Argument #1 is not an array in /home/pl
  aysports/public_html/magento2/vendor/magento/framework/App/DeploymentConfig
  /Writer.php on line 135


setup:di:compile
```
![image](https://user-images.githubusercontent.com/12158143/37635985-7e561774-2bd5-11e8-80f7-da21f3755fc7.png)

**Comment by magento-engcom-team** [comment]
@mikeguy ,please refer to the [Community Forums](https://community.magento.com/) or the [Magento Stack Exchange](https://magento.stackexchange.com/) site for advice or general discussion about this issue. The GitHub issue tracker is intended for Magento Core technical issues only.  Dotmailer is not core codbase funcionally

**Comment by mikeguy** [comment]
@magento-engcom-team 
Ok , so this error code when I perform the compile has to do with the dotmailer? You have confirmed ?
It crashes when i perform magento setup:di:compile so I am confused how this is not from core?

I feel that you have just quickly skimmed through the first line I wrote without looking at everything else. There are many other tickets like this that you guys follow up on.  setup:di:compile is core...no? 
For example Issue  [#12097](https://github.com/magento/magento2/issues/12097)

**Comment by mikeguy** [comment]
OK for anyone else who ever runs into this issue the problem was that the env.php file got corrupted somehow when disabling the module (in /app/etc).
I did not have a backup, so I found one available online: https://github.com/emizentech/magento2/blob/master/app/etc/env.php.sample
I used that to build it again and once complete I copied it into the /app/etc directory.

I still think this is a core issue as somehow this file got overwritten, and it shouldn't have... but anyhow.
